backend/import/do_import.py

```diff
@@ -152,65 +160,137 @@ def handle_this(obj):
             writer(data)
         return handle_this
 
-    ou_handler = handler(get_orgunit_data, orgunit_writer)
-    employee_handler = handler(get_employee_data, employee_writer)
+    ou_handler = handler(get_orgunit_data, writer.write_unit)
+    employee_handler = handler(get_employee_data, writer.write_employee)
 
-    p = Pool(10)
+    p = multiprocessing.dummy.Pool(jobs)
     # First, org units
     p.map(ou_handler, (mo_api.OrgUnit(ou['uuid']) for ou in ous))
-    p.close()
-    p.join()
 
-    p = Pool(10)
     # Now, employees
     p.map(employee_handler, (mo_api.Employee(e['uuid']) for e in employees))
+
+    # Finally, wait
     p.close()
     p.join()
 
 
-def file_writer(directory, field_name='uuid'):
-    """Return a function that writes data to a given directory.
-
-    The data must be JSON and is  stored in a file name taken from the
-    data's field as indicated by the ``field_name`` parameter."""
-
-    base_dir = os.path.abspath(
-        os.path.join(os.path.dirname(__file__), '..', 'var')
-    )
-    target_dir = os.path.join(base_dir, directory)
+class AbstractWriter(abc.ABC):
+    def __init__(self, verbose=False):
+        self.__verbose = verbose
+
+    @abc.abstractmethod
+    def write(self, type, data):
+        pass
+
+    @abc.abstractmethod
+    def clean(self, type):
+        pass
+
+    def _do_write(self, type, data):
+        if self.__verbose:
+            log("{}: {}", type, data['uuid'])
+
+        self.write(type, data)
+
+    def write_unit(self, data):
+        self._do_write('departments', data)
+
+    def write_employee(self, data):
+        self._do_write('employees', data)
+
+
+class FileWriter(AbstractWriter):
+    def __init__(self, base_dir, verbose=False):
+        super().__init__(verbose)
+        self.base_dir = pathlib.Path(base_dir)
+
+    def clean(self):
+        if self.base_dir.exists():
+            shutil.rmtree(str(self.base_dir))
+
+    def write(self, type, data):
```

Comment:
It's generally a bad idea to use `type` as an identifier, given that it's a reserved word.